acir/src/circuit/opcodes.rs

```diff
@@ -6,11 +6,24 @@ use crate::serialization::{read_n, read_u16, read_u32, write_bytes, write_u16, w
 use crate::BlackBoxFunc;
 use serde::{Deserialize, Serialize};
 
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Hash, Copy, Default)]
+pub struct BlockId(u32);
+
+/// Operation on a block
+/// We can either write or read at a block index
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct BlockOp {
```

Comment:
Do these blocks in acvm have any correspondence to ssa blocks? It seems they are meant to be similar to the MemOps in ssa, I think we should give it a similar name in that case.

Reply:
Indeed, they have nothing to do with ssa blocks, they represent an abstract block of data where you can do read and write operations on it, aka the MemOps in ssa. I am open to change the naming.

Reply:
I have renamed it to MemOp. At the end (i.e when this PR is merged), ssa will be changed to use this structure instead of the ssa's MemOp.